nexus/src/external_api/http_entrypoints.rs

```diff
@@ -352,10 +352,33 @@ pub(crate) fn external_api() -> NexusApiDescription {
         api.register(device_auth::device_auth_confirm)?;
         api.register(device_auth::device_access_token)?;
 
-        api.register(probe_list)?;
-        api.register(probe_view)?;
-        api.register(probe_create)?;
-        api.register(probe_delete)?;
+        Ok(())
+    }
+
+    fn register_experimental<T>(
+        api: &mut NexusApiDescription,
+        endpoint: T,
+    ) -> Result<(), String>
+    where
+        T: Into<ApiEndpoint<Arc<ServerContext>>>,
+    {
+        let mut ep: ApiEndpoint<Arc<ServerContext>> = endpoint.into();
+        // only one tag is allowed
+        ep.tags = vec![String::from("hidden")];
+        let experimental = String::from("/experimental");
+        if !ep.path.starts_with(&experimental) {
+            ep.path = experimental + &ep.path;
+        }
```

Comment:
I'm a little bit unsure of this kind of conditional magic to be honest -- I'd rather either expect that the path begins with `/experimental`, or expect that it doesn't and add it.

Reply:
This has been updated as we discussed in chat to be an unconditional update.